src/vunnel/providers/mariner/parser.py

```diff
@@ -0,0 +1,196 @@
+
+from vunnel.providers.mariner.model import Definition, RpminfoObject, RpminfoState, RpminfoTest
+from vunnel.utils.vulnerability import FixedIn, Vulnerability
+
+if TYPE_CHECKING:
+    from collections.abc import Generator
+
+    from vunnel.workspace import Workspace
+
+LTE = "less than or equal"
+
+
+class MarinerXmlFile:
+    def __init__(self, oval_file_path: str, logger: logging.Logger):
+        parser_config = ParserConfig(
+            fail_on_converter_warnings=False,
+            fail_on_unknown_attributes=False,
+            fail_on_unknown_properties=False,
+        )
+        xml_parser = XmlParser(config=parser_config)
+        root = etree.parse(oval_file_path)
+        nsmap = etree.XPath("/*")(root)[0].nsmap
+        default = nsmap[None]
+        nsmap["default"] = default
+        del nsmap[None]
+        self.logger = logger
+
+        # Go element by element to have flexible error handling
+        self.definitions = []
+        for def_element in etree.XPath("//default:definition", namespaces=nsmap)(root):
+            try:
+                element_bytes = etree.tostring(def_element)
+                definition = xml_parser.from_string(element_bytes.decode(), Definition)
+                self.definitions.append(definition)
+            except Exception as ex:
+                self.logger.warning(f"skipping definition element in {oval_file_path} due to {ex}")
+                pass
+
+        self.tests_by_id = {}
+        for test_element in etree.XPath("//linux-def:rpminfo_test", namespaces=nsmap)(root):
+            try:
+                element_bytes = etree.tostring(test_element)
+                test_obj = xml_parser.from_string(element_bytes.decode(), RpminfoTest)
+                self.tests_by_id[test_obj.id] = test_obj
+            except Exception as ex:
+                self.logger.warning(f"skipping rpminfo_test element in {oval_file_path} due to {ex}")
+        self.objects_by_id = {}
+        for obj_element in etree.XPath("//linux-def:rpminfo_object", namespaces=nsmap)(root):
+            try:
+                element_bytes = etree.tostring(obj_element)
+                obj = xml_parser.from_string(element_bytes.decode(), RpminfoObject)
+                self.objects_by_id[obj.id] = obj
+            except Exception as ex:
+                self.logger.warning(f"skipping rpminfo_object element in {oval_file_path} due to {ex}")
+
+        self.states_by_id = {}
+        for state_element in etree.XPath("//linux-def:rpminfo_state", namespaces=nsmap)(root):
+            try:
+                element_bytes = etree.tostring(state_element)
+                state_obj = xml_parser.from_string(element_bytes.decode(), RpminfoState)
+                self.states_by_id[state_obj.id] = state_obj
+            except Exception as ex:
+                self.logger.warning(f"skipping rpminfo_object element in {oval_file_path} due to {ex}")
+        self.mariner_version = oval_file_path.split("-")[-2]  # TODO: less brittle way?
```

Comment:
Do we need a less brittle way here?

The `TODO` was meant for this PR, but I don't see a way to make this better; if the Mariner XML files break their naming convention, this will break, but I can't think of anything that won't.